### PR TITLE
Add `linkerd inject --manual` test into install_test.go

### LINKED ISSUE #2744: Create an integration test for manual injection
As of #2733 the auto-injector is enabled and by default all injection is performed by it, unless we use `linkerd inject --manual` which outputs the injected YAML.
As a result most of the integration tests are relying on auto-injection.
We should enhance `install_test.go` with a scenario dealing with manual injection.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #2744

### CODE REVIEW

File: test/install_test.go [hunk 1]
```diff
@@ -49,12 +52,15 @@ var (
...
 
+	injectorErr = `.* linkerd-proxy-injector-.*-.* proxy-injector time=".*" level=warning msg="failed to retrieve replicaset from indexer, retrying with get request .*-%s/smoke-test-.*-.*: replicaset\.apps \\"smoke-test-.*-.*\\" not found"`
+
 	// Linkerd commonly logs these errors during testing, remove these once
 	// they're addressed: https://github.com/linkerd/linkerd2/issues/2453
 	knownControllerErrorsRegex = regexp.MustCompile(strings.Join([]string{
 		`.* linkerd-controller-.*-.* tap time=".*" level=error msg="\[.*\] encountered an error: rpc error: code = Canceled desc = context canceled"`,
 		`.* linkerd-web-.*-.* web time=".*" level=error msg="Post http://linkerd-controller-api\..*\.svc\.cluster\.local:8085/api/v1/Version: context canceled"`,
-		`.* linkerd-proxy-injector-.*-.* proxy-injector time=".*" level=warning msg="failed to retrieve replicaset from indexer, retrying with get request .*-smoke-test/smoke-test-.*-.*: replicaset\.apps \\"smoke-test-.*-.*\\" not found"`,
+		fmt.Sprintf(injectorErr, nsSmokeAuto),
```
Comment (siggy) [inline test/install_test.go]: how about just `retrying with get request .*-smoke-test.*/smoke-test-.*-.*`

File: test/install_test.go [hunk 2]
```diff
@@ -299,48 +305,88 @@ func TestDashboard(t *testing.T) {
...
-	}
 
-	out, err = TestHelper.KubectlApply(out, prefixedNs)
-	if err != nil {
-		t.Fatalf("kubectl apply command failed\n%s", out)
-	}
+	testCases := []struct {
+		ns          string
+		autoinject  bool
+		annotations map[string]string
```
Comment (siggy) [inline test/install_test.go]: i can think of 3 ways to inject:
1. set auto-inject on namespace ✔️
2. `linkerd inject --manual` ✔️
3. regular `linkerd inject` on a non-auto-injected namespace (maybe add this one?)

File: test/install_test.go [hunk 3]
```diff
@@ -299,48 +305,88 @@ func TestDashboard(t *testing.T) {
...
+				if err != nil {
+					t.Fatalf("linkerd inject command failed: %s\n%s", err, out)
+				}
+
+				err = TestHelper.ValidateOutput(injectReport, "inject.report.golden")
+				if err != nil {
+					t.Fatalf("Received unexpected output\n%s", err.Error())
+				}
+
+				out, err = TestHelper.KubectlApply(out, prefixedNs)
```
Comment (siggy) [inline test/install_test.go]: this deploys a second time for the test case, consider modifying the code above so we only call `KubectlApply` once per test?

File: test/testdata/inject.report.golden [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+deployment "smoke-test-terminus" injected
```
Comment (siggy) [inline test/testdata/inject.report.golden]: 👍

File: test/install_test.go [hunk 5]
```diff
@@ -49,12 +49,14 @@ var (
...
+
 	// Linkerd commonly logs these errors during testing, remove these once
 	// they're addressed: https://github.com/linkerd/linkerd2/issues/2453
 	knownControllerErrorsRegex = regexp.MustCompile(strings.Join([]string{
 		`.* linkerd-controller-.*-.* tap time=".*" level=error msg="\[.*\] encountered an error: rpc error: code = Canceled desc = context canceled"`,
 		`.* linkerd-web-.*-.* web time=".*" level=error msg="Post http://linkerd-controller-api\..*\.svc\.cluster\.local:8085/api/v1/Version: context canceled"`,
-		`.* linkerd-proxy-injector-.*-.* proxy-injector time=".*" level=warning msg="failed to retrieve replicaset from indexer, retrying with get request .*-smoke-test/smoke-test-.*-.*: replicaset\.apps \\"smoke-test-.*-.*\\" not found"`,
+		`.* linkerd-proxy-injector-.*-.* proxy-injector time=".*" level=warning msg="failed to retrieve replicaset from indexer, retrying with get request .*-smoke-test-.*-.*: replicaset\.apps \\"smoke-test-.*-.*\\" not found"`,
```
Comment (siggy) [inline test/install_test.go]: i think this should be `.*-smoke-test.*/smoke-test-.*-.*`

File: test/install_test.go [hunk 6]
```diff
@@ -49,12 +49,14 @@ var (
 		"linkerd-proxy-injector": {1, []string{"proxy-injector"}},
 	}
 
+	injectorErr = `.* linkerd-proxy-injector-.*-.* proxy-injector time=".*" level=warning msg="failed to retrieve replicaset from indexer, retrying with get request .*-%s/smoke-test-.*-.*: replicaset\.apps \\"smoke-test-.*-.*\\" not found"`
```
Comment (siggy) [inline test/install_test.go]: remove?

Comment (alpeb) [reply test/install_test.go]: oops

File: test/install_test.go (hunk 5 above)
Comment (alpeb) [reply test/install_test.go]: Indeed! :-)

**Pull request merged**